src/pages/CreateClub/CreateClub.tsx

```diff
@@ -48,50 +56,89 @@ export const CreateClub: React.FC = () => {
     }
   };
 
-  const onChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
+  const onChange = ({
+    target,
+  }: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     setFormData((prev) => ({
       ...prev,
       [target.name]: target.value,
     }));
   };
 
+  const handleBack = () => {
+    setCurrentStep(1);
+  };
+
+  if (isSubmitted) {
+    return (
+      <Layout>
+        <div>
+          <h1>Club created successfully!</h1>
+          <p className={style.description}>
+            view your brand new{' '}
+            <Link to={`/clubs/${id}`} className={style.link}>
+              {formData.name}
+            </Link>
+          </p>
+        </div>
+        ;
+      </Layout>
+    );
+  }
+
   return (
     <Layout>
       <div className={style.createClubPage}>
-        {isSubmitted ? (
-          <div>
-            <h1>Club created successfully!</h1>
-            <p>
-              {' '}
-              view your brand new{' '}
-              <Link to={`/clubs/${id}`}>{formData.name}</Link>
-            </p>
-          </div>
-        ) : (
+        <div className={style.formSection}>
           <FormWrapper
             title="Create Club"
             onSubmit={onSubmit}
-            submitText="Create"
+            submitText={currentStep === 1 ? 'Next' : 'Create'}
           >
-            <FormInput
-              type="text"
-              name="name"
-              placeholder="Club Name"
-              required
-              onChange={onChange}
-              value={formData.name}
-            />
-
-            <FormInput
-              type="text"
-              name="description"
-              placeholder="Club Description"
-              required
-              onChange={onChange}
-              value={formData.description}
-            />
+            <div className={style.stepIndicator}>Step {currentStep}/2</div>
+
+            {currentStep === 1 ? (
```

Comment:
Yours is probably safer. I was thinking about:

![CleanShot 2025-01-15 at 10 43 15](https://github.com/user-attachments/assets/941a59b6-2a07-46ff-89ab-1f0885d64e25)